plugin-scalebar/src/main/java/com/mapbox/pluginscalebar/ScaleBarOption.java

```diff
@@ -0,0 +1,307 @@
+  }
+
+  /**
+   * Build a scale bar widget instance with current option settings.
+   *
+   * @return The built ScaleBarWidget instance.
+   */
+  ScaleBarWidget build() {
+    ScaleBarWidget scaleBarWidget = new ScaleBarWidget(context);
+    scaleBarWidget.setBarHeight(barHeight);
+    scaleBarWidget.setBorderWidth(borderWidth);
+    scaleBarWidget.setMarginLeft(marginLeft);
+    scaleBarWidget.setMarginTop(marginTop);
+    scaleBarWidget.setTextBarMargin(textBarMargin);
+    scaleBarWidget.setMetricUnit(isMetricUnit);
+    scaleBarWidget.setRefreshInterval(refreshInterval);
+    scaleBarWidget.setPrimaryColor(primaryColor);
+    scaleBarWidget.setSecondaryColor(secondaryColor);
+    scaleBarWidget.setTextColor(textColor);
+    scaleBarWidget.setTextSize(textSize);
+    return scaleBarWidget;
+  }
+
+  /**
+   * Set plugin's minimum refresh interval,
+   * default value is {@link #REFRESH_INTERVAL_DEFAULT}.
+   *
+   * @param refreshInterval the min refresh interval, in millisecond.
+   * @return this
+   */
+  public ScaleBarOption setRefreshInterval(int refreshInterval) {
+    this.refreshInterval = refreshInterval;
+    return this;
+  }
+
+  /**
+   * Set the text color on scale bar,
+   * default value is android.R.color.black.
+   *
+   * @param textColor the text color on scale bar.
+   * @return this.
+   */
+  public ScaleBarOption setTextColor(@ColorRes int textColor) {
+    this.textColor = context.getResources().getColor(textColor);
+    return this;
+  }
+
+  /**
+   * Set the primary color of the scale bar, will be used to draw odd index blocks,
+   * default value is android.R.color.black.
+   *
+   * @param primaryColor the primary color of the scale bar.
+   * @return this.
+   */
+  public ScaleBarOption setPrimaryColor(@ColorRes int primaryColor) {
+    this.primaryColor = context.getResources().getColor(primaryColor);
+    return this;
+  }
+
+  /**
+   * Set the secondary color of the scale bar, will be used to draw even index blocks,
+   * default value is android.R.color.white.
+   *
+   * @param secondaryColor the secondaryColor color of the scale bar.
+   * @return this.
+   */
+  public ScaleBarOption setSecondaryColor(@ColorRes int secondaryColor) {
+    this.secondaryColor = context.getResources().getColor(secondaryColor);
+    return this;
+  }
+
+  /**
+   * Set the margin between scale bar and the top of mapView.
+   *
+   * @param marginTop the margin between scale bar and the top of mapView, in pixel.
+   * @return this.
+   */
+  public ScaleBarOption setMarginTop(float marginTop) {
+    this.marginTop = marginTop;
+    return this;
+  }
+
+  /**
+   * Set the margin between scale bar and the top of mapView.
+   *
+   * @param marginTop the margin between scale bar and the top of mapView, in dp.
+   * @return this.
+   */
+  public ScaleBarOption setMarginTop(@DimenRes int marginTop) {
+    this.marginTop = convertDpToPixel(context.getResources().getDimension(marginTop));
```

Comment:
No need to call `convertDpToPixel`. `getDimension` call already returns the value in pixels.